### DISCUSSION TITLE
First Use Server Side Rendering Second At Navigation next/link Use Client Side Rendering (struggling 2 day🤦‍♂️)

### DISCUSSION THREAD

**Discussion by sromexs** [vercel/next.js#14123] [Help]
Hi
I'm working an next js these days and i want to do something cool. when user make http request to my page i (for SEO) i want page render in Server Side and show all data at ones (i want fetch data at Server Side). but when page load completed and user navigate between pages using `next/lint` i want data render in Client Side (i want fetch data in Client Side). with `getInitialProps` we can do this let me show you an example.
```javascript
function Index({ posts }) {
  const [postsArr, setPostsArr] = useState(posts);

  useEffect(() => {
    async function fetchNewPosts() {
      const res = await fetch("https://getpost.com/");
      setPostsArr(await res.json());
    }

    if (posts.length === 0) {
      fetchNewPosts();
    }
  }, []);

  return (
    <>
      <Nav />
      <p>Hello World May I Pick Banana!</p>
      <pre>{JSON.stringify(postsArr, null, 4)}</pre>
    </>
  );
}

Index.getInitialProps = async ({ req }) => {
  let posts = [];
  if (req) {
    const res = await fetch("https://getpost.com/");
    posts = await res.json();
  }
  return { posts };
};

export default Index;
```
in this approach if there is request we fetch data at first place (Server Side Rendering) but if we navigating to this page with `next/link` it fetches data from client side (Client Side Rendering).
But with these new data fetching methods  `getStaticProps` and `getServerSideProps` i cant do above behavior.
is there any one to help me to do above behavior with these new methods ? ( `getStaticProps` and `getServerSideProps`). Thanks for reading my question.

**Comment by rafaelalmeidatk** [comment]
This is not possible without a hack, we have an open issue similar to what you are asking, consider upvoting it: https://github.com/vercel/next.js/issues/13910
